### PR TITLE
WebRTC Demo Styling

### PR DESCRIPTION
**Pull request by manonxrray**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -4,41 +4,55 @@
...
-          <input type="text" id="server" class="form-control login input-lg" placeholder="Server" required>
-	        <button class="btn btn-lg btn-primary btn-block btn-flat login" id="submit-login" type="submit">Login</button>
-          <div id="auth-error" class="alert alert-danger" role="alert"></div>
+          <div class="form-group">
+            <label for="username">Username</label>
+            <input type="text" id="username" name="username" class="login" placeholder="votrenom@mail.com" required autofocus>
+            <label for="password">Password</label>
+            <input type="password" id="password" name="password" class="login" placeholder="********" required>
+            <label for="server">Server URI</label>
+            <input type="text" id="server" name="server" class="login" placeholder="votreserver.io" required>
```
Comment (sboily) [inline index.html:25]: English

File: index.html [hunk 2]
```diff
@@ -4,41 +4,55 @@
...
         <form class="center" id="login-form">
           <img class="wazo-logo" src="img/wazo-logo.svg" />
-          <input type="text" id="username" class="form-control login input-lg" placeholder="Login" required autofocus>
-          <input type="password" id="password" class="form-control login input-lg" placeholder="Password" required>
-          <input type="text" id="server" class="form-control login input-lg" placeholder="Server" required>
-	        <button class="btn btn-lg btn-primary btn-block btn-flat login" id="submit-login" type="submit">Login</button>
-          <div id="auth-error" class="alert alert-danger" role="alert"></div>
+          <div class="form-group">
+            <label for="username">Username</label>
+            <input type="text" id="username" name="username" class="login" placeholder="votrenom@mail.com" required autofocus>
```
Comment (sboily) [inline index.html:21]: english

File: src/dialer.js [hunk 3]
```diff
@@ -4,40 +4,69 @@ let inConference = false;
...
 // let sessionIdsInMerge = [];
 
-const setFullName = async () => {
-  const user = await Wazo.getApiClient().confd.getUser(Wazo.Auth.getSession().uuid);
-  const fullName = user.firstName ? `${user.firstName} ${user.lastName}` : user.username;
+/*
+  Permet de récupérer les informations de l'utilisateur connecté 
+  Cette fonction ne sert plus à rien normalement puisqu'on a besoin d'afficher seulement
+  le prénom de l'utilisateur.
+*/
```
Comment (sboily) [inline src/dialer.js:11]: Comment in english please

File: src/dialer.js [hunk 4]
```diff
@@ -4,40 +4,69 @@ let inConference = false;
...
+//     : user.username;
 
-  $('#user').html(`${fullName} (${user.lines[0].extensions[0].exten})`);
+//   $("#user").html(`${fullName} (${user.lines[0].extensions[0].exten})`);
+// };
+
+/*
+  Permet de récupérer les informations de l'utilisateur connecté
+  puis d'afficher le prénom de l'utilisateur
+*/
```
Comment (sboily) [inline src/dialer.js:27]: english

File: src/dialer.js [hunk 5]
```diff
@@ -4,40 +4,69 @@ let inConference = false;
...
 
-const setMainStatus = status => {
-  $('#dialer .status').html(status);
+/* Permet d'indiquer le statut de l'appel. */
+const setMainStatus = (status) => {
+  $(".status").html(status);
 };
 
-const getNumber = callSession => callSession.realDisplayName || callSession.displayName || callSession.number;
+/* ??? récupère le numéro j'imagine mais euh */
```
Comment (sboily) [inline src/dialer.js:42]: english

File: src/dialer.js [hunk 6]
```diff
@@ -4,40 +4,69 @@ let inConference = false;
...
+    Wazo.Auth.getSession().uuid
+  );
+  const name = user.firstName;
+
+  $(".greeter").html(`Hello ${name} 👋`);
 };
 
-const setMainStatus = status => {
-  $('#dialer .status').html(status);
+/* Permet d'indiquer le statut de l'appel. */
```
Comment (sboily) [inline src/dialer.js:37]: english

File: src/dialer.js [hunk 7]
```diff
@@ -4,40 +4,69 @@ let inConference = false;
...
 
-const getNumber = callSession => callSession.realDisplayName || callSession.displayName || callSession.number;
+/* ??? récupère le numéro j'imagine mais euh */
+const getNumber = (callSession) =>
+  callSession.realDisplayName || callSession.displayName || callSession.number;
 
-const getStatus = callSession => {
+/*
+  Permet de récupérer le statut de l'appel
+*/
```
Comment (sboily) [inline src/dialer.js:48]: english

**Review by sboily — CHANGES_REQUESTED** [review]

**Comment by sboily** [comment]
Looks like the dialer is not available on this PR.

File: src/dialer.js (hunk 7 above)
Comment (manonxrray) [reply src/dialer.js:49]: I pushed my code so Jeff can take a look and I forgot to erase my comments. I wrote them for myself to untangle each function and understand the code better. I used to write them in english, I didn't mean these ones to be read 😅

File: index.html (hunk 2 above)
Comment (manonxrray) [reply index.html]: I will change that right away !

File: styles.css [hunk 8]
```diff
@@ -17,111 +15,283 @@
...
-  height: 200px;
-  display: block;
-  float: left;
-  margin-right: 30px;
+/* MAIN */
+
+.btn-primary:active,
+.btn-primary.hover,
+.btn-primary:hover {
+    background: #eebf3f !important;
```
Comment (manuquentin) [inline styles.css:23]: Can you use 2 spaces for indentation, not 4.

File: styles.css [hunk 9]
```diff
@@ -17,111 +15,283 @@
...
-  display: block;
-  width: 200px;
-  height: 200px;
+.btn-flat {
+    -webkit-box-shadow: none;
+    -moz-box-shadow: none;
+    box-shadow: none;
+    background-image:none;
+    background: #515454;
+    border:0;
```
Comment (manuquentin) [inline styles.css:33]: Missing space after the colon

File: src/dialer.js [hunk 10]
```diff
@@ -215,9 +256,9 @@ function resetMainDialer(status) {
 
     resetMainDialer('');
     updateDialers();
-  }
+  };
 
-  dialer.off('submit').on('submit', function (e) {
+  dialer.off('submit').on('submit', (e) => {
```
Comment (manuquentin) [inline src/dialer.js:261]: `e => {` (same line 283)

File: src/dialer.js [hunk 11]
```diff
@@ -153,7 +193,8 @@ function startConference() {
 
 function endConference() {
   inConference = false;
-  const sessionToUnmerge = Object.values(sessions).filter(session => sessionIdsInMerge.indexOf(session.getId()) !== -1);
+  const sessionToUnmerge = Object.values(sessions)
```
Comment (manuquentin) [inline src/dialer.js:196]: Can you change the eslint config to set the max-len to 1️⃣2️⃣0️⃣ ?

File: src/authentication.js [hunk 12]
```diff
@@ -66,11 +68,11 @@ const launchPhone = async () => {
     if (session) {
       return onLogin(session);
     }
-  } catch(e) {
+  } catch (e) {
     displayAuthError(e);
   }
 
-  openLogin();
+  return openLogin();
```
Comment (manuquentin) [inline src/authentication.js:75]: `openLogin` doesn't returns anything and is not async. There is not need to return the result of this function here.

File: index.html [hunk 13]
```diff
@@ -4,104 +4,173 @@
...
-        </form>
-      </div>
-      <div class="hidden-xs hidden-sm col-md-9 two col">
-        <div class="col-md-12"></div>
+        </div>      
+      </form>
+    </div>
+    <div class="hidden-xs hidden-sm col-md-9 two col">
+      <div class="col-md-12 right-quote">
+        <p>A certain king had a beautiful garden, and in the garden stood a tree.</p>
```
Comment (manuquentin) [inline index.html:33]: Is it a @bogue's weird sentence ?

File: index.html [hunk 14]
```diff
@@ -4,104 +4,173 @@
...
+    <!-- Nav header when logged in -->
+    <nav class="navbar navbar-expand navbar-dark bg-wazo fixed-top">
+      <a class="navbar-brand" href="#">
+        <img src="img/wazo-platform-logo.svg" height="40" alt="logo">
+      </a>
+      <!-- <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
+        <span class="navbar-toggler-icon"></span>
+      </button> -->
+      <a href="#" id="logout">Logout</a>
+      <!-- <div class="collapse navbar-collapse justify-content-end" id="navbarNavAltMarkup">
```
Comment (manuquentin) [inline index.html:55]: Can you remove commented out code ?

File: src/dialer.js [hunk 15]
```diff
@@ -1,15 +1,11 @@
+/* eslint-disable max-len */
```
Comment (manuquentin) [inline src/dialer.js:1]: Should be done in a `.eslintrc` file

File: index.html [hunk 16]
```diff
@@ -143,7 +143,7 @@ <h3 class="greeter"></h3>
           
       </form>
 
-      <p id="status"></p>
+      <p id="status" class=""></p>
```
Comment (manuquentin) [inline index.html:146]: Why an empty class here ?

File: index.html (hunk 13 above)
Comment (manonxrray) [reply index.html]: SORRY I just noticed I had comments :( Not I think it's Sylvain's or anyone who did the mock ups !

File: .gitignore [hunk 17]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (manuquentin) [inline .gitignore:1]: Every time you create a file, don't forget to add a empty line at the end.
It's a convention that ease the reading when we use commands like `cat`.

File: index.html [hunk 18]
```diff
@@ -4,104 +4,236 @@
...
-              <div class="dropdown-menu dropdown-menu-right dropdown-default"
-                aria-labelledby="navbarNavAltMarkup">
-                <a class="dropdown-item" href="#" id="logout">Logout</a>
-              </div>
-            </li>
-          </ul>
+    <!-- Nav header when logged in -->
+    <nav class="navbar navbar-expand navbar-dark bg-wazo fixed-top">
+      <a class="navbar-brand" href="#">
+        <img src="img/wazo-platform-logo.svg" height="40" alt="logo">
```
Comment (manuquentin) [inline index.html:55]: Missing tag closing (eg: `/`)

File: index.html [hunk 19]
```diff
@@ -4,104 +4,236 @@
...
-              </a>
-              <div class="dropdown-menu dropdown-menu-right dropdown-default"
-                aria-labelledby="navbarNavAltMarkup">
-                <a class="dropdown-item" href="#" id="logout">Logout</a>
-              </div>
-            </li>
-          </ul>
+    <!-- Nav header when logged in -->
+    <nav class="navbar navbar-expand navbar-dark bg-wazo fixed-top">
+      <a class="navbar-brand" href="#">
```
Comment (manuquentin) [inline index.html:54]: Should we link to wazo-platform.org ?

File: index.html [hunk 20]
```diff
@@ -4,104 +4,236 @@
...
+        <img src="img/wazo-platform-logo.svg" height="40" alt="logo">
+      </a>
+      <!-- On call dialer -->
+
+      <!--
```
Comment (manuquentin) [inline index.html:59]: Please remove commented out code

File: index.html [hunk 21]
```diff
@@ -4,104 +4,236 @@
...
+            </div>  
+          </div>  
+        </div>        
+      </div>
+
+      <div
+        id="calls-handler"
+        class="btn-group-vertical" 
+        style="
+        position:absolute;
```
Comment (manuquentin) [inline index.html:194]: Inline css

File: src/dialer.js [hunk 22]
```diff
@@ -1,43 +1,48 @@
...
 let currentAtxfer;
+let countDown;
+let timerId;
 // let sessionIdsInMerge = [];
 
-const setFullName = async () => {
-  const user = await Wazo.getApiClient().confd.getUser(Wazo.Auth.getSession().uuid);
-  const fullName = user.firstName ? `${user.firstName} ${user.lastName}` : user.username;
+const setGreeter = async () => {
+  const user = await Wazo.getApiClient().confd.getUser(
```
Comment (manuquentin) [inline src/dialer.js:10]: Can be written in one line

File: src/dialer.js [hunk 23]
```diff
@@ -1,43 +1,48 @@
...
 
-const setMainStatus = status => {
-  $('#dialer .status').html(status);
+const setMainStatus = (status) => {
+  $('#status').html(status);
+  setTimeout(() => $('#status').html('&nbsp;'), 3000);
 };
 
-const getNumber = callSession => callSession.realDisplayName || callSession.displayName || callSession.number;
+const getNumber = (callSession) => callSession.realDisplayName || callSession.displayName || callSession.number;
```
Comment (manuquentin) [inline src/dialer.js:23]: The parenthesis aren't required for single argument

File: src/dialer.js [hunk 24]
```diff
@@ -1,43 +1,48 @@
...
+      return `Calling ${number}...`;
     case Wazo.Phone.SessionState.Established:
-      return 'On call with : ' + number;
+      return `On call with : ${number}`;
     case Wazo.Phone.SessionState.Terminated:
-      return 'Call canceled by ' + number;
+      return `Call canceled by ${number}`;
     default:
-      return 'Unknown status: ' + callSession.sipStatus;
+      return `Unknown status: ${callSession.sipStatus}`;
```
Comment (manuquentin) [inline src/dialer.js:45]: `Call in ${callSession.sipStatus} status` because `Unknown` can be seen as messy.

File: src/dialer.js [hunk 25]
```diff
@@ -47,236 +52,203 @@ const initializeWebRtc = () => {
...
   Wazo.Phone.on(Wazo.Phone.ON_CALL_HELD, onSessionUpdate);
-  Wazo.Phone.on(Wazo.Phone.ON_CALL_UNHELD, onSessionUpdate);
+  Wazo.Phone.on(Wazo.Phone.ON_CALL_RESUMED, onSessionUpdate);
   Wazo.Phone.on(Wazo.Phone.ON_CALL_MUTED, onSessionUpdate);
   Wazo.Phone.on(Wazo.Phone.ON_CALL_UNMUTED, onSessionUpdate);
-  Wazo.Phone.on(Wazo.Phone.ON_REINVITE, (session, request, updatedCalleeName) => {
-    const callSession = sessions[session.id];
-    if (callSession) {
-      currentSession.realDisplayName = updatedCalleeName;
+  Wazo.Phone.on(
```
Comment (manuquentin) [inline src/dialer.js:65]: We aren't in the 80 column line of code era anymore :)
```
Wazo.Phone.on(Wazo.Phone.ON_REINVITE, (session, request, updatedCalleeName) => {
```

File: src/dialer.js [hunk 26]
```diff
@@ -47,236 +52,203 @@ const initializeWebRtc = () => {
...
+    Wazo.Phone.ON_REINVITE,
+    (session, request, updatedCalleeName) => {
+      const callSession = sessions[session.id];
+      if (callSession) {
```
Comment (manuquentin) [inline src/dialer.js:69]: You can write:
```
if (!callSession) {
  return;
}
```
To avoid extra indentation

File: src/dialer.js [hunk 27]
```diff
@@ -47,236 +52,203 @@ const initializeWebRtc = () => {
...
   delete sessions[callSession.getId()];
+  $('.buttons').removeClass('buttons-video');
+  $('#incoming-modal').hide();
+  $('#dialer').show();
 
   // Current session terminated ?
   if (currentSession && currentSession.getId() === callSession.getId()) {
     // Remaining session ? take first
-    currentSession = Object.keys(sessions).length ? sessions[Object.keys(sessions)[0]] : null;
+    currentSession = Object.keys(sessions).length
```
Comment (manuquentin) [inline src/dialer.js:117]: can fit in 1 line :)

File: src/dialer.js [hunk 28]
```diff
@@ -47,236 +52,203 @@ const initializeWebRtc = () => {
...
 }
 
 function accept(callSession, withVideo) {
-  // Hold current session if exists
+  console.log(`accepting ${getNumber(callSession)} ${withVideo ? 'withVideo' : 'withoutVideo'}`);
+  // Hold current session & creates the multiple calls handler if exists 
   if (currentSession && !inConference) {
     hold(currentSession);
+    
+    const currentNumber = getNumber(currentSession);
```
Comment (manuquentin) [inline src/dialer.js:144]: Are `currentNumber` and `newNumber` used ?

File: src/dialer.js [hunk 29]
```diff
@@ -47,236 +52,203 @@ const initializeWebRtc = () => {
...
   }
 
   Wazo.Phone.accept(callSession, withVideo);
-
-  onCallAccepted(callSession, withVideo);
 }
 
 function unhold(callSession) {
-  Wazo.Phone.unhold(callSession);
+  console.log(`resuming ${getNumber(callSession)}`);
```
Comment (manuquentin) [inline src/dialer.js:152]: Should be removed after the debug done :)

File: src/dialer.js [hunk 30]
```diff
@@ -47,236 +52,203 @@ const initializeWebRtc = () => {
...
+    const currentNumber = getNumber(currentSession);
+    const newNumber = getNumber(callSession);
   }
 
   Wazo.Phone.accept(callSession, withVideo);
-
-  onCallAccepted(callSession, withVideo);
 }
 
 function unhold(callSession) {
```
Comment (manuquentin) [inline src/dialer.js:151]: Can be rewritten like :
```
const unhold = callSession => Wazo.Phone.resume(callSession);
```
Same for methods below

File: src/dialer.js [hunk 31]
```diff
@@ -47,236 +52,203 @@ const initializeWebRtc = () => {
...
-  const muteButton = $('.mute', newDialer);
-  const unmuteButton = $('.unmute', newDialer);
-  const mergeButton = $('.merge', newDialer).html('Add to merge');
-  const unmergeButton = $('.unmerge', newDialer).html('Remove from merge');
-  const atxferButton = $('.atxfer', newDialer);
-  const transferButton = $('.transfer', newDialer);
-  const videoButton = $('.video-call', newDialer);
-
-  $('.form-group', newDialer).hide();
+  const hangupButton = $('.hangup', newScene);
```
Comment (manuquentin) [inline src/dialer.js:233]: Maybe we can write something like :
```
const getSceneElements = (...elements) => elements.map(element => $(element, newScene));
const [unholdButton, holdButton, ...] = getSceneElements('.unhold', '.hold', ...);
```

File: src/dialer.js [hunk 32]
```diff
@@ -286,20 +258,64 @@ function addDialer(callSession, withVideo) {
   unmergeButton.hide();
   atxferButton.hide();
   transferButton.hide();
+  reduceVideoButton.hide();
```
Comment (manuquentin) [inline src/dialer.js:261]: Same here :
```
const hideElements = (...elements) => elements.forEarch(element => element.hide());
// ...
hideElements(unmergeButton, atxferButton, ...);
```

File: src/dialer.js [hunk 33]
```diff
@@ -286,20 +258,64 @@ function addDialer(callSession, withVideo) {
...
   if (withVideo) {
     videoContainer.show();
+    videoContainer.addClass('background-videocall');
+    $('.buttons').addClass('buttons-video');
+    reduceVideoButton.show();
+    reduceLocalVideo.show();
+
+    // Reduce & Expand Remote video screen
+    
+    reduceVideoButton.on('click', e => {
```
Comment (manuquentin) [inline src/dialer.js:280]:
```
const bindClick = (element, cb) => {
  element.off('click');
  element.on('click', evt => {
    evt.preventDefault;
    cb(evt);
  });
}

bindClick(reduceVideoButton, () => {
  $('.remote video').addClass('reduce-video');
    reduceVideoButton.hide();
    expandVideoButton.show();
});
```

File: src/dialer.js [hunk 34]
```diff
@@ -286,20 +258,64 @@ function addDialer(callSession, withVideo) {
...
+      $('.local video').removeClass('reduce-video');
+      expandLocalVideo.hide();
+      reduceLocalVideo.show();
+    })
+
 
     // Local video
     const localStream = Wazo.Phone.getLocalVideoStream(callSession);
-    const localVideo = $('video.local', newDialer)[0];
+    const localVideo = $('.local video', newScene)[0];
```
Comment (manuquentin) [inline src/dialer.js:313]: Can be moved where we declare you elements line 249 (same for `$remoteVideo`)

File: src/dialer.js [hunk 35]
```diff
@@ -286,20 +258,64 @@ function addDialer(callSession, withVideo) {
...
     localVideo.srcObject = localStream;
     localVideo.play();
 
     // Remote video
-    const $remoteVideo = $('video.remote', newDialer);
+    const $remoteVideo = $('.remote video', newScene);
```
Comment (manuquentin) [inline src/dialer.js:318]: It's a good practice to prefix jquery elements with `$` but it should be applied to all elements or none :)

File: src/dialer.js [hunk 36]
```diff
@@ -392,45 +396,143 @@ function addDialer(callSession, withVideo) {
   });
 
   transferButton.show();
-  transferButton.off('click').on('click', function (e) {
+  transferButton.off('click').on('click', (e) => {
     e.preventDefault();
 
-    const target = prompt("Phone number transfer?");
+    const target = prompt('Phone number transfer?');
```
Comment (manuquentin) [inline src/dialer.js:402]: `Phone number to transfer to ?` ?

File: src/dialer.js [hunk 37]
```diff
@@ -392,45 +396,143 @@ function addDialer(callSession, withVideo) {
...
+
+
+const getElapsedTimeInSeconds = (startTime) => {
+  if (!startTime) {
+    return 0;
+  }
+  return Math.trunc((Date.now() - startTime) / 1000);
+};
+
+const formatTime = secondsElapsed  => {
```
Comment (manuquentin) [inline src/dialer.js:438]: Can we move all these kind of method to a `utils.js` file ?

File: styles.css [hunk 38]
```diff
@@ -1,127 +1,474 @@
...
+}
+
+/* LOADER */
+
+#loader {
+  position : fixed;
+  z-index: 9999;
+  background-image: url('./img/back.jpeg');
+  background-size: cover;
+  top : 0px;
```
Comment (manuquentin) [inline styles.css:86]: `0` no need for `px`

File: styles.css [hunk 39]
```diff
@@ -1,127 +1,474 @@
...
+}
+
+h3.contact {
+  color: #fff;
+  text-align: center;
+}
+
+/* ON CALL DIALER */
+
+/*
```
Comment (manuquentin) [inline styles.css:453]: Commented out code

**Review by bogue — APPROVED** [review]

**Pull request merged**